data/src/test/java/org/apache/iceberg/data/TestDataFileIndexStatsFilters.java

```diff
@@ -94,24 +93,23 @@ public void createTableAndData() throws IOException {
             .filter(rec -> rec.getField("category").equals("even"))
             .collect(Collectors.toList());
 
-    this.dataFile = FileHelpers.writeDataFile(table, Files.localOutput(temp.newFile()), records);
     this.dataFileWithoutNulls =
         FileHelpers.writeDataFile(
             table,
-            Files.localOutput(temp.newFile()),
+            Files.localOutput(temp),
```

Comment:
```suggestion
            Files.localOutput(File.createTempFile("junit", null, temp.toFile())),
```
same as above, you can take a look at `DeleteReadTests` as a reference. Please also update all the other places in this class here